app/src/main/java/com/teamagam/gimelgimel/app/map/esri/MapDragEventsEmitterTouchListenerDecorator.java

```diff
@@ -0,0 +1,112 @@
+package com.teamagam.gimelgimel.app.map.esri;
+
+import android.view.MotionEvent;
+import android.view.View;
+import com.esri.android.map.MapOnTouchListener;
+import com.esri.android.map.MapView;
+import com.teamagam.gimelgimel.app.map.MapDragEvent;
+import com.teamagam.gimelgimel.domain.map.entities.geometries.PointGeometry;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.subjects.Subject;
+
+class MapDragEventsEmitterTouchListenerDecorator implements View.OnTouchListener {
+
+  private MapOnTouchListener mDecorated;
+  private MapDragEventsEmitter mMapDragEventsEmitter;
+
+  public MapDragEventsEmitterTouchListenerDecorator(MapOnTouchListener decorated,
+      EsriGGMapView esriGGMapView,
+      Subject<MapDragEvent> subject,
+      BiFunction<Float, Float, PointGeometry> screenToGround) {
+    mDecorated = decorated;
+    mMapDragEventsEmitter = new MapDragEventsEmitter(esriGGMapView, subject, screenToGround);
+  }
+
+  @Override
+  public boolean onTouch(View v, MotionEvent event) {
+    mMapDragEventsEmitter.onTouch(v, event);
+    return mDecorated.onTouch(v, event);
+  }
+
+  private class MapDragEventsEmitter extends MapOnTouchListener {
+
+    private final Subject<MapDragEvent> mMapDragEventSubject;
+    private final BiFunction<Float, Float, PointGeometry> mScreenToGrounder;
```

Comment:
It's more complex than that since it needs to know the current map's CRS to do so.